File: mobile/android/speechcore/src/main/kotlin/com/shinsoku/mobile/speechcore/TranscriptCommitPlanner.kt
package com.shinsoku.mobile.speechcore

object TranscriptCommitPlanner {
    fun plan(text: String, profile: VoiceInputProfile): VoiceInputCommit =
        VoiceInputCommit(
            text = text + when (profile.commitSuffixMode) {
                CommitSuffixMode.None -> ""
                CommitSuffixMode.Space -> " "
                CommitSuffixMode.Newline -> "\n"
            },
        )
}

File: mobile/nativecore/src/transcript_commit_planner.cpp
#include "shinsoku/nativecore/transcript_commit_planner.hpp"

namespace shinsoku::nativecore {

std::string plan_transcript_commit(const std::string& text, CommitSuffixMode suffix_mode) {
    switch (suffix_mode) {
        case CommitSuffixMode::None:
            return text;
        case CommitSuffixMode::Space:
            return text + " ";
        case CommitSuffixMode::Newline:
            return text + "\n";
    }
    return text;
}

}  // namespace shinsoku::nativecore
